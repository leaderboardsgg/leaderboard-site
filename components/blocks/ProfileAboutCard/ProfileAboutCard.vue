<template>
  <div class="profile-about-card">
    <div class="border-b border-gray-300">
      <h2 class="text-xl font-bold">About</h2>
    </div>

    <div>
      <p class="text-black">{{ bio }}</p>
      <div class="mt-5">
        <CoreSocialButtons :socials="socials" />
      </div>
    </div>
  </div>
</template>

<script>
import { defineComponent } from '@nuxtjs/composition-api';

export default defineComponent({
  props: {
    bio: {
      default: () =>
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed eget dapibus libero. Morbi ultricies varius accumsan.',
      type: String,
    },
    socials: {
      default: () => [
        {
          icon: 'dscrd',
          name: 'Discord',
          url: 'https://discord.com/',
        },
        {
          icon: 'twch',
          name: 'Twitch',
          url: 'https://twitch.tv/',
        },
        {
          icon: 'twtr',
          name: 'Twitter',
          url: 'https://twitter.com/',
        },
        {
          icon: 'yt',
          name: 'Youtube',
          url: 'https://youtube.com/',
        },
      ],
      type: Array,
    },
  },
});
</script>

<style scoped>
.profile-about-card {
  @apply hidden md:flex flex-col border rounded border-gray-300;
}

.profile-about-card > div {
  @apply p-5;
}
</style>
