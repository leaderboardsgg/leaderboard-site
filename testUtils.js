import Vue from 'vue';
import { RouterLinkStub } from '@vue/test-utils';
import userEvent from '@testing-library/user-event';
import { render } from '@testing-library/vue';

export const NuxtLinkStub = Vue.component('NuxtLinkStub', {
  extends: RouterLinkStub,
  render(h) {
    const temp = h(
      this.tag || 'a',
      {
        attrs: {
          ...this.$vnode.data.attrs,
          ...this.$attrs,
          href: this.to,
        },
      },
      this.$slots.default,
    );

    return temp;
  },
});

export const stubbedRender = (Component, options) => {
  return render(Component, {
    stubs: { NuxtLink: NuxtLinkStub, ...options?.stubs },
    ...options,
  });
};

export * from '@testing-library/vue';
export * from '@vue/test-utils';
export { userEvent as fireEvent };
