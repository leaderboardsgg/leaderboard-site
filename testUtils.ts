import Vue, { ComponentOptions } from 'vue';
import { RouterLinkStub, VueClass } from '@vue/test-utils';
import userEvent from '@testing-library/user-event';
import { render, RenderOptions } from '@testing-library/vue';

export const NuxtLinkStub = Vue.component<
  { tag: string; to: string },
  unknown,
  unknown
>('NuxtLinkStub', {
  extends: RouterLinkStub,
  render(h) {
    const temp = h(
      this.tag || 'a',
      {
        attrs: {
          ...this.$vnode.data?.attrs,
          ...this.$attrs,
          href: this.to,
        },
      },
      this.$slots.default,
    );

    return temp;
  },
});

export const stubbedRender = (
  Component: VueClass<Vue> | ComponentOptions<Vue>,
  options?: RenderOptions<Vue>,
) => {
  return render(Component, {
    stubs: { NuxtLink: NuxtLinkStub, ...options?.stubs },
    ...options,
  });
};

export * from '@testing-library/vue';
export * from '@vue/test-utils';
export { userEvent as fireEvent };
