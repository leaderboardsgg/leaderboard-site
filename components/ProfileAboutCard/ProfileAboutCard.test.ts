import ProfileAboutCard from './ProfileAboutCard.vue';
import { stubbedRender } from '@/testUtils';

describe('<ProfileAboutCard />', () => {
  it('should render without crashing', () => {
    const { unmount } = stubbedRender(ProfileAboutCard);

    unmount();
  });

  it('renders correctly', () => {
    const { container } = stubbedRender(ProfileAboutCard);

    expect(container.firstChild).toMatchSnapshot();
  });
});
