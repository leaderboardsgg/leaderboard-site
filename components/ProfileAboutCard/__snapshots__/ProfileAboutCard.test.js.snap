// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileAboutCard /> renders correctly 1`] = `
<div
  class="profile-about-card"
>
  <div
    class="border-b border-gray-300"
  >
    <h2
      class="text-xl font-bold"
    >
      About
    </h2>
  </div>
   
  <div>
    <p
      class="text-black"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed eget dapibus libero. Morbi ultricies varius accumsan.
    </p>
     
    <div
      class="mt-5"
    >
      <div
        class="social-button--container"
      >
        <a
          alt="Discord"
          class="social-button"
          href="https://discord.com/"
        >
          
    dscrd
  
        </a>
        <a
          alt="Twitch"
          class="social-button"
          href="https://twitch.tv/"
        >
          
    twch
  
        </a>
        <a
          alt="Twitter"
          class="social-button"
          href="https://twitter.com/"
        >
          
    twtr
  
        </a>
        <a
          alt="Youtube"
          class="social-button"
          href="https://youtube.com/"
        >
          
    yt
  
        </a>
      </div>
    </div>
  </div>
</div>
`;
